### ISSUE TITLE
Unstructured Covariance Documented but Missing

### ISSUE THREAD

**Issue by billdenney** [glmmTMB/glmmTMB#761]
I'm wanting a model with unstructured covariance.  I found the `us()` covariance structure in the vignette here: https://cran.r-project.org/web/packages/glmmTMB/vignettes/covstruct.html

When I looked at the help for `glmmTMB()`, I saw that it appeared to no longer exist.  It would be helpful if either `us()` were added back or it were removed from the vignette.

**Comment by bbolker** [comment]
`us()` certainly still exists. We may have neglected to document it **because it is the default**, i.e. `(f|g)` is synonymous with `us(f|g)` ...

**Comment by billdenney** [comment]
Hmm.  Alrighty, then I have an issue with my code that I need to track down some other way.  (I'm getting an error that the function `us` does not exist, but the data I'm working on is proprietary, so I can't immediately share a reprex.)

I suppose that this morphs the issue into a request to add `us()` to the `glmmTMB()` help page.

**Comment by bbolker** [comment]
Can you share an anonymized `glmmTMB()` call?  Are you working in a funny environment (e.g. passing data and formulas among different environments and then evaluating or re-evaluating them?)

**Comment by bbolker** [comment]
Let me rephrase my comment above slightly: a `us()` function does *not* exist, but `glmmTMB` shouldn't need it to.  The tags indicating covariance structures (from https://cran.r-project.org/web/packages/glmmTMB/vignettes/covstruct.html : `us()`, `toep()`, `ar1()` etc.) should never need to be evaluated as functions in the course of a regular `glmmTMB` evaluation.

**Comment by bbolker** [comment]
any further guesses/clues/reprexes?

**Comment by billdenney** [comment]
Here we go...

The vignette suggests that `us()` should work (even if it is the default) here: https://cran.r-project.org/web/packages/glmmTMB/vignettes/covstruct.html#the-unstructured-covariance

But, the code below does not work due to not finding the `us()` function.

``` r
library(glmmTMB)
#> Warning: package 'glmmTMB' was built under R version 4.1.1
library(tidyverse)
#> Warning: package 'tibble' was built under R version 4.1.1
#> Warning: package 'readr' was built under R version 4.1.1
my_data <-
  crossing(
    A=factor(1:10),
    B=factor(1:10)
  ) %>%
  mutate(
    C=exp(rnorm(100))
  )

foo <- glmmTMB(C~A*B, data=my_data)
#> Warning in fitTMB(TMBStruc): Model convergence problem; non-positive-definite
#> Hessian matrix. See vignette('troubleshooting')
#> Warning in fitTMB(TMBStruc): Model convergence problem; function evaluation
#> limit reached without convergence (9). See vignette('troubleshooting')
bar <- glmmTMB(C~us(A*B), data=my_data)
#> Error in us(A * B): could not find function "us"
```

<sup>Created on 2021-09-14 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by bbolker** [comment]
This is a weird model specification.  `us()` is only supposed to be used in the context of a random effect, in a model term with an effect and a grouping variable, e.g. 

```r
library(glmmTMB)
data("sleepstudy", package="lme4")
m1 <- glmmTMB(Reaction ~ 1 + (Days|Subject), sleepstudy)
m2 <- glmmTMB(Reaction ~ 1 + us(Days|Subject), sleepstudy)
all.equal(VarCorr(m1), VarCorr(m2))
## [1] TRUE
```

What was your goal in specifying `us(A*B)` , i.e. what model were you trying to specify?  Something about correlations of the residuals, or about the correlations of the fixed effects, or ... ??

**Comment by billdenney** [comment]
My goal is that each term of the `A*B` fixed effects and interactions have a separate variance estimate.  (I did forget to specify the random effects, but that seems like a different issue.)

That model structure with a different residual variability for each of the fixed effect interactions (and a random intercept by subject) is the typical model used in clinical drug development.

**Comment by bbolker** [comment]
OK, I think that would be something like 

```r
bar <- glmmTMB(C~A*B, dispformula = ~A*B, data=my_data)
```
? (Doesn't converge, of course, but gives rise to with 100 fixed-effect parameters and 100 dispersion parameters)

**Comment by bbolker** [comment]
Any luck?

**Comment by bbolker** [comment]
Closing; feel free to re-open.
